service/history/execution/mutable_state_util_test.go

```diff
@@ -316,3 +327,252 @@ func TestConvertWorkflowRequests(t *testing.T) {
 
 	assert.ElementsMatch(t, expectedOutputs, convertWorkflowRequests(inputs))
 }
+
+func TestCreatePersistenceMutableState(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockShardContext := shard.NewMockContext(ctrl)
+	logger := testlogger.New(t)
+	mockEventsCache := events.NewMockCache(ctrl)
+	mockDomainCache := cache.NewMockDomainCache(ctrl)
+	mockShardContext.EXPECT().GetClusterMetadata().Return(constants.TestClusterMetadata).Times(2)
+	mockShardContext.EXPECT().GetEventsCache().Return(mockEventsCache)
+	mockShardContext.EXPECT().GetConfig().Return(config.NewForTest())
+	mockShardContext.EXPECT().GetTimeSource().Return(clock.NewMockedTimeSource())
+	mockShardContext.EXPECT().GetMetricsClient().Return(metrics.NewClient(tally.NoopScope, metrics.History))
+	mockShardContext.EXPECT().GetDomainCache().Return(mockDomainCache)
+
+	builder := newMutableStateBuilder(mockShardContext, logger, constants.TestLocalDomainEntry)
+	builder.pendingActivityInfoIDs[0] = &persistence.ActivityInfo{}
+	builder.pendingTimerInfoIDs["some-key"] = &persistence.TimerInfo{}
+	builder.pendingSignalInfoIDs[0] = &persistence.SignalInfo{}
+	builder.pendingChildExecutionInfoIDs[0] = &persistence.ChildExecutionInfo{}
+	builder.bufferedEvents = []*types.HistoryEvent{{}}
+	builder.updateBufferedEvents = []*types.HistoryEvent{{}}
+	builder.versionHistories = &persistence.VersionHistories{CurrentVersionHistoryIndex: 0, Histories: []*persistence.VersionHistory{}}
+	builder.pendingRequestCancelInfoIDs[0] = &persistence.RequestCancelInfo{}
+	builder.decisionTaskManager.(*mutableStateDecisionTaskManagerImpl).HasInFlightDecision()
+	builder.executionInfo.DecisionStartedID = 1
+
+	mutableState := CreatePersistenceMutableState(builder)
+	assert.NotNil(t, mutableState)
+}
+
+func TestGetChildExecutionDomainName(t *testing.T) {
+	t.Run("nonempty domain ID", func(t *testing.T) {
+		childInfo := &persistence.ChildExecutionInfo{DomainID: testdata.DomainID}
+		mockDomainCache := cache.NewMockDomainCache(gomock.NewController(t))
+		expected := testdata.DomainName
+		mockDomainCache.EXPECT().GetDomainName(childInfo.DomainID).Return(expected, nil)
+		name, err := GetChildExecutionDomainName(childInfo, mockDomainCache, constants.TestLocalDomainEntry)
+		assert.Nil(t, err)
```

Comment:
`assert.NoError()` is a better candidate for error-checking.
Also, I recommend using `require` in this case since most probably you're not interested in checking for value if err != nil; in fact, in lots of cases it will even panic.

Reply:
changed